corehq/apps/app_manager/tests/test_suite_session_endpoints.py

```diff
@@ -125,11 +130,13 @@ def test_multiple_session_endpoints(self):
                     <argument id="parent_id"/>
                     <argument id="case_id"/>
                     <stack>
-                    <push>
-                        <command value="'m1-f0'"/>
-                        <datum id="parent_id" value="$parent_id"/>
-                        <datum id="case_id" value="$case_id"/>
-                    </push>
+                        <push>
+                            <command value="'m0'"/>
+                            <command value="'m1'"/>
+                            <command value="'m1-f0'"/>
+                            <datum id="parent_id" value="$parent_id"/>
+                            <datum id="case_id" value="$case_id"/>
+                        </push>
                     </stack>
                 </endpoint>
```

Comment:
@shubham1g5 does this endpoint look right to you?  I was a little surprised to see the `datum`s all at the end, but I didn't really touch that part of the logic from end-of-form navigation and this is what it gave me.

Reply:
@esoergel Yeah this does look suspicious. I think the order is not important currently for mobile's use case of session end point as mobile applies all the stack ops together without checking whether the intermediary steps are valid or not. Although this works a bit differently on Formplayer where Formplayer make sure that even the [intermediary steps](https://github.com/dimagi/formplayer/blob/master/src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java#L49) are correct. We have also talked about switching to Formplayer style of processing stack ops on mobile as well to manage navigation better in future.  I would say it's probably worth a look to see what's causing the wrong ordering here and see if it would be an easy thing to fix or not. Otherwise I think @orangejenny is likely to come across this issue soon when doing the similar session endpoint work on Web Apps.

Reply:
Thanks, I dug into it a bit more and it looks like an artifact of some of the weirdness in this file.  I commented below on the line that causes the problem.  Looks like this code has two different ways of inserted parent module data, so I extracted the one used by end-of-form-navigation and used that for session endpoints, which gives better-looking results.  At least they're consistent with EoF navigation.

I think that's as far as I want to go with this PR, as I think anything else would risk a behavior change, which if necessary, I'd like to do in a standalone PR for more thorough consideration and an easier revert path.

Reply:
that makes total sense, thanks for digging further!